From 8579b172d5644f655a3ae772bae2e053981f17a5 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Sat, 20 Jan 2024 14:43:24 -0500
Subject: [PATCH 12/34] Allow forcing rotation on external display

Check if the property ro.boot.android.force_rotation_on_external_displays is set and use that for external displays
0=0
1=90
2=180
3-270
---
 .../android/server/display/DisplayManagerService.java    | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/services/core/java/com/android/server/display/DisplayManagerService.java b/services/core/java/com/android/server/display/DisplayManagerService.java
index b394d6b49d65..697339ff7041 100644
--- a/services/core/java/com/android/server/display/DisplayManagerService.java
+++ b/services/core/java/com/android/server/display/DisplayManagerService.java
@@ -537,6 +537,15 @@ public final class DisplayManagerService extends SystemService {
                 }
                 final DisplayPowerController dpc = mDisplayPowerControllers.get(
                         logicalDisplay.getDisplayIdLocked());
+                
+                // Check if the property ro.boot.android.force_rotation_on_external_displays is set and use that for external displays
+                String forceRotationOnExternalDisplays = SystemProperties.get("ro.boot.android.force_rotation_on_external_displays","0");
+                if (device.getDisplayDeviceInfoLocked().type==Display.TYPE_EXTERNAL) {
+                    if (forceRotationOnExternalDisplays != "0") {
+                        device.getDisplayDeviceInfoLocked().rotation=Integer.valueOf(forceRotationOnExternalDisplays);
+                    }
+                }
+                
                 if (dpc == null) {
                     return;
                 }
-- 
2.43.0

